scipio/src/sys/uring.rs

```diff
@@ -914,23 +914,14 @@ impl Reactor {
     pub(crate) fn wait(
         &self,
         wakers: &mut Vec<Waker>,
-        timeout: Option<Duration>,
-        timer_expiration: Option<Duration>,
+        preempt: Option<Duration>,
+        spin: Option<Duration>,
```

Comment:
This feels wrong. The fact that `timeout` and `timer_expiration` were separate here used to tell us if we are ok to go to sleep: we could sleep with no timeout, but we'd have to set a timer for `timer_expiration` if there was any.

(this whole thing is underdocumented, which is my fault).

Now, how will you be able to tell that you can sleep ?
Spin could signal that, but you still need to tell the no spin case from the spin so it works with both kinds of executors